californium-core/src/main/java/org/eclipse/californium/core/network/stack/ReliabilityLayer.java

```diff
@@ -364,6 +365,15 @@ public void run() {
 					LOGGER.log(Level.FINE, "Timeout: retransmission limit reached, exchange failed, message: {0}", message);
 					exchange.setTimedOut();
 					message.setTimedOut(true);
+					OptionSet options = message.getOptions();
+					boolean isBlockwise = options.hasBlock1() || options.hasBlock2();
+					if (message instanceof Request && isBlockwise) {
+					    Request request = exchange.getRequest();
+					    if (request != null && request.getMID() != message.getMID()) {
+					        // Notifying the original request about timeout
+					        request.setTimedOut(true);
+					    }
```

Comment:
I am not sure if we should invoke `request.setTimedOut()` on the original request *in addition* to `message.setTimedOut()` or *instead* of it ...
In the former case we might end up with two listeners being invoked. But maybe that's what is intended if somebody registers a `MessageListener` on the *intermediary* GET of the blockwise transfer.